### PR TITLE
Feature/invite clubs to tournaments

### PR DESCRIPTION
**Pull request by xmk0987**
Did ticket: 

https://trello.com/c/tpHbwIzu/69-inviting-players-by-club

and fixed a bug: 

https://trello.com/c/jsGYRymD/90-editing-club-name-not-working

### CODE REVIEW

File: backend/server/src/controllers/tournamentController.ts [hunk 1]
```diff
@@ -75,6 +75,7 @@ export class TournamentController extends Controller {
     @Path() tournamentId: ObjectIdString,
     @Body() teamData: { name: string }
   ): Promise<Tournament> {
+    console.log("Team added");
```
Comment (metsoila) [inline backend/server/src/controllers/tournamentController.ts:78]: Intentional?

File: backend/server/src/services/tournamentService.ts [hunk 2]
```diff
@@ -304,6 +311,27 @@ export class TournamentService {
       });
     }
 
+    // Verify the password if the tournament is password-protected
+    if (tournament.passwordEnabled) {
```
Comment (metsoila) [inline backend/server/src/services/tournamentService.ts:315]: image

does this work with Lint

File: frontend/src/components/modules/Profile/ProfileInfo.tsx [hunk 3]
```diff
@@ -93,6 +93,7 @@ const ProfileInfo: React.FC = () => {
 
   const onSubmit = async (data: EditUserRequest): Promise<void> => {
     try {
+      console.log("update user", data);
```
Comment (metsoila) [inline frontend/src/components/modules/Profile/ProfileInfo.tsx:96]: intentional?

File: frontend/src/components/modules/Tournaments/InviteButton.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  };
+
+  const handleClubChange = (event: SelectChangeEvent<string[]>): void => {
+    setSelectedClubs(event.target.value as string[]);
+  };
+
+  const handleInvite = async (): Promise<void> => {
+    try {
+      if (selectedClubs.length > 0) {
+        console.log("Inviting players from clubs:", selectedClubs);
```
Comment (metsoila) [inline frontend/src/components/modules/Tournaments/InviteButton.tsx:45]: intentional to leave?

File: frontend/src/components/modules/Tournaments/Signup/Signup.tsx [hunk 5]
```diff
@@ -45,9 +47,26 @@ const Signup: React.FC = (): ReactElement => {
...
+        return;
+      }
+
+      // Prepare the signup data conditionally based on passwordEnabled
+      const signupData: { playerId: string; password?: string } = {
+        playerId: userId
+      };
+
+      // Include the password only if passwordEnabled is true
+      if (tournament.passwordEnabled) {
```
Comment (metsoila) [inline frontend/src/components/modules/Tournaments/Signup/Signup.tsx:62]: image

lint workings?

**Review by metsoila — APPROVED** [review]

**Review by rhourani — APPROVED** [review]
Good job!

File: backend/server/src/services/tournamentService.ts (hunk 2 above)
Comment (xmk0987) [reply backend/server/src/services/tournamentService.ts:315]: its weird this format fails with the ! operator but for truthy values it passes

File: frontend/src/components/modules/Tournaments/InviteButton.tsx (hunk 4 above)
Comment (xmk0987) [reply frontend/src/components/modules/Tournaments/InviteButton.tsx:45]: ill delete them on next feature

**Pull request merged**